src/nvim/eval/typval.c

```diff
@@ -1164,6 +1164,35 @@ void callback_free(Callback *callback)
   callback->type = kCallbackNone;
 }
 
+/// Copy a callback into a typval_T.
+void callback_put(Callback *cb, typval_T *tv)
+  FUNC_ATTR_NONNULL_ALL
+{
+  if (cb->type == kCallbackPartial) {
+    tv->v_type = VAR_PARTIAL;
+    tv->vval.v_partial = cb->data.partial;
+    cb->data.partial->pt_refcount++;
+  } else if (cb->type == kCallbackFuncref) {
+    tv->v_type = VAR_FUNC;
+    tv->vval.v_string = vim_strsave(cb->data.funcref);
+    func_ref(cb->data.funcref);
+  }
```

Comment:
what happens when `kCallbackNone` ? is `tv` expected to be in a pre-initialized state (like number 0 or v:null)?

Reply:
I don't know based on the current code. I'll push a new commit to force v:null.